src/main/webapp/index.html

```diff
@@ -58,44 +58,42 @@ <h2>Product Photo</h2>
             Choose an image with the product you are searching for, and we'll compile a shopping 
             list of similar products for you.
           </p>
-          <p><a class="btn btn-secondary" href="#" role="button">Upload product photo »</a></p>
+          <p><button class="btn btn-secondary">Upload product photo »</button></p>
         </div><!-- /.col-lg-4 -->
         <div class="col-lg-4">
           <i class="fa fa-list"></i>
           <h2>Shopping List</h2>
           <p>
             The shopping list may be hand-written.
           </p>
-          <p><a class="btn btn-secondary" href="#" role="button">Upload list photo »</a></p>
+          <p><button class="btn btn-secondary">Upload list photo »</button></p>
         </div><!-- /.col-lg-4 -->
         <div class="col-lg-4">
           <i class="fa fa-barcode"></i>
           <h2>Barcode</h2>
           <p>
             Upload a barcode
           </p>
-          <p><a class="btn btn-secondary" href="#" role="button" id="upload-barcode" onclick="fileUploadDisplay()">Upload barcode photo »</a></p>
+          <p><button class="btn btn-secondary" id="upload-barcode" onclick="openFileUploadDialog()">Upload barcode photo »</button></p>
         </div><!-- /.col-lg-4 -->
 
         <!-- UPLOAD DIV -->
         <div class="bg-model">
           <div class= "model-content">
-            <button class="close" onclick="closeDiv()"> + </button>
+            <button class="close" onclick="closeFileUploadDialog()"> + </button>
              <h1> Upload barcode </h1>
-             <br>
                 <form id="upload-barcode-form" class="hidden" method="POST" enctype="multipart/form-data">
                   <input type="file" name="barcode" required>
-                  <br>
                   <button>Submit</button>
                 </form>
             </div>
         </div>
 
       </div><!-- /.row -->
 
-      <div id="history">
+      <div id ="history">
```

Comment:
Nit: remove space before `=`.

For some reason, in HTML you don't use spaces around `=` but you do use it in JavaScript and Java.

Reply:
I wasn't aware. Thanks for letting me know.